File: tools/samples/ruby/lib/box.rb
# a leftover comment
class Box
  def initialize(value)
    @value = value
  end

  def value
    read_value
  end

  def classify(n)
    if n > 10
      if n > 20
        if n > 30
          if n > 40
            "huge"
          else
            "big"
          end
        else
          "mid"
        end
      else
        "small"
      end
    else
      "tiny"
    end
  end

  private

  def read_value
    @value
  end

  def unused_helper
    1
  end
end

File: tools/samples/erlang/src/box.erl
% a leftover comment
-module(box).

-export([new/1, value/1, tag/1, classify/1, raw_value/1]).

new(Value) ->
    {box, Value}.

value(Box) ->
    box_store:fetch(Box).

raw_value({box, Value}) ->
    Value.

tag(Box) ->
    case raw_value(Box) of
        empty ->
            empty;
        Value ->
            {tagged, Value}
    end.

classify(N) when is_integer(N) ->
    if
        N > 10 ->
            if
                N > 20 ->
                    if
                        N > 30 ->
                            if
                                N > 40 -> huge;
                                true -> big
                            end;
                        true -> mid
                    end;
                true -> small
            end;
        true -> tiny
    end.
